Draft 1: Experimental features blog (#4061)
opensearch-project/project-website: _posts/2026-01-15-experimental-features.md

---
layout: post
title: "OpenSearch’s Hidden Powers - Experimental Features"
category: blog
authors:
    - nateboot
date: 2026-01-30
categories:
  - technical-posts
meta_keywords: 
meta_description: 
---

> I was recently writing about our latest experimental features - the implementation of "Apache Arrow Flight." I thought about some ot the other features that we have that are technically "experimental" features that may not be getting the attention they deserve. I thought I'd take the chance and tell you about some of them. 

Reviewer: ```suggestion
I was [recently writing](https://opensearch.org/blog/opensearch-and-apache-arrow-a-tour-of-the-archery-range/?_gl=1*3orc3i*_up*MQ..*_ga*NDM1Mzk5Mzc1LjE3NzE1MDE5Mzg.*_ga_BQV14XK08F*czE3NzE1MDE5MzgkbzEkZzAkdDE3NzE1MDE5MzgkajYwJGwwJGgxNDMyNDcwMjA.) about our latest experimental features---the implementation of Apache Arrow and Arrow Flight---and thought about some of the other OpenSearch features that are technically "experimental" but may not be getting the attention they deserve.
```

> I've learned that something we take very seriously here at OpenSearch is our ability to innovate and explore new ways of breaking down problems into smaller, more easily digestible steps. Sometimes doing so requires a bit of experimentation. We have a number of features that are marked as experimental so you can test tomorrow’s innovation today. Of course they’re cool and exciting, but border on risky in their experimental form. This frontier between stability and risk is exactly where innovation lives. We need your help to bring them into the forefront. 

Reviewer: ```suggestion
I've learned that something the OpenSearch Project takes very seriously is innovating and exploring new ways of breaking down problems into smaller, more easily digestible steps. Sometimes doing so requires a bit of experimentation. We have a number of features that are marked as experimental so that you can test tomorrow’s innovation today. Of course they’re cool and exciting, but they may border on risky in their experimental form. This frontier between stability and risk is exactly where innovation lives. We need your help to bring them to the forefront. 
```

> **What Are They?**

Reviewer: ```suggestion
**What are experimental features?**
```

These are features that aren’t quite production ready, but represent significant innovation. They are best used in development environments, as not all of them have been in the oven long enough to be considered fully baked.

**Why try something that isn’t done yet?**

For one, early access to innovation. New algorithms, UI changes, performance optimizations, and bleeding edge ML features are some things that tend to get hidden behind feature flags. This is also an area in which the feedback of the community literally shapes OpenSearch’s future. Your contribution in testing these is driving development and innovation forward. The light bulb did not come about via continuous improvement of the candle.

**Where are you hiding these things?**

> When features are hiding behind a feature flag, what we’re talking about is a config option inside of either `opensearch.yml` or `opensearch_dashboards.yml`.  How about a few examples?

Reviewer: ```suggestion
An experimental feature hidden behind a feature flag can be accessed through a configuration option in either `opensearch.yml` or `opensearch_dashboards.yml`. Let's look at a few examples.
```

**Here are some suggestions on what to try out!**

**The “Workspace” feature** reconfigures the UI so that you’re able to tailor the on-screen elements in a way that is specifically targeted towards the use case you’re implementing.  Observability use case will rely heavily on visualizing health and performance through logs and metrics.

[Image: Image.jpg]

To enable the workspace feature, drop a few lines into your `opensearch_dashboards.yml` file.

```workspace.enabled: true
uiSettings:
    overrides:
        "home:useNewHomePage": true
```
If your cluster has the security plugin installed, multi-tenancy must be disabled. Drop this into your dashboards configuration also.

```
opensearch_security.multitenancy.enabled: false
```
**The “Distributed Tracing” feature**

This feature has been experimental since version 2.10, and is a great experiment for people with application tracing use cases. Watch the full journey of a request from its inception to completion in a distributed environment. There’s a couple of edits needed to enable this one.

> To your `opensearch.yml` file, add `opensaerch.experimental.feature.telemetry.enabled=true.`

Reviewer: ```suggestion
- To your `opensearch.yml` file, add `opensearch.experimental.feature.telemetry.enabled=true`.
```
